utils/unittest/network/p2p.go

```diff
@@ -0,0 +1,22 @@
+package network
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/onflow/flow-go/network/p2p/p2pnode"
+)
+
+// StopNodes stop all nodes in the input slice
+func StopNodes(t *testing.T, nodes []*p2pnode.Node) {
```

Comment:
There are new versions of these that support the new `Component` interface of libp2pnode
https://github.com/onflow/flow-go/blob/master/network/p2p/internal/p2pfixtures/fixtures.go#L230

Reply:
These are removed now see @yhassanzadeh13 comment. Also we don't need to explicitly stop the libp2p nodes in the middle ware test. This is because the middle ware will control the lifecycle of the node.

Reply:
> we don't need to explicitly stop the libp2p nodes in the middle ware test. This is because the middle ware will control the lifecycle of the node.

That used to be true, but now that libp2p nodes are components, their lifecycle is handled explicitly by cancelling the context passed to `Start()`. It looks like middleware_test is missing a call to stop the nodes. that should probably go in a `TearDown` method